### ISSUE TITLE
Bug: `no-unsafe-values` does not report hexadecimal integers that contain the digit `e`

### ISSUE THREAD

**Issue by electrohyun** [eslint/json#266]
### Environment

```
ESLint version: v10.8.0
@eslint/json version: v2.0.1
Node version: v22.22.3
npm version: v10.8.2
Operating System: StackBlitz WebContainer (linux)
```

### Which language are you using?

```
json5
```

### What did you do?

<details>
<summary>Configuration</summary>

```js
// eslint.config.js
import json from "@eslint/json";

export default [
	{
		files: ["**/*.json5"],
		plugins: { json },
		language: "json/json5",
		rules: { "json/no-unsafe-values": "error" },
	},
];
```
</details>

Since a JSON5 document holds a single value, I used four files: two for the case I am reporting and two as controls.

```js
// a-hex-with-e.json5
0xe0000000000000

// d-hex-precision-loss.json5
0xdeadbeefdeadbeef

// b-hex-without-e.json5   (control)
0x20000000000000

// c-decimal.json5         (control)
9007199254740993
```

### What did you expect to happen?

I expected all four files to be reported.

All four values are outside the safe integer range (`±2^53-1`) that this rule checks. `0xe0000000000000` is `63050394783186944` and `0xdeadbeefdeadbeef` is `16045690984833335023`, both well above `Number.MAX_SAFE_INTEGER` (`9007199254740991`). The two control values are outside that range as well, and they are reported correctly.

`0xdeadbeefdeadbeef` also loses precision when read as a JavaScript number: it becomes `16045690984833335296`, so the value silently changes.

### What actually happened?

Only the two control files are reported. Both hexadecimal values that contain the digit `e` are silently accepted:

```
❯ npx eslint .

/home/projects/stackblitz-starters-hqubvayr/b-hex-without-e.json5
  1:1  error  The integer '0x20000000000000' is outside the safe integer range  json/no-unsafe-values

/home/projects/stackblitz-starters-hqubvayr/c-decimal.json5
  1:1  error  The integer '9007199254740993' is outside the safe integer range  json/no-unsafe-values

✖ 2 problems (2 errors, 0 warnings)
```

Neither `a-hex-with-e.json5` nor `d-hex-precision-loss.json5` appears in the output at all.

### Link to Minimal Reproducible Example

```
https://stackblitz.com/edit/stackblitz-starters-hqubvayr
```

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Hello. I looked into where this comes from, and here is what I found.

**Where the bug is**

In `src/rules/no-unsafe-values.js`, the `Number` handler decides whether a value was "intended to be an integer" by testing the raw text for `.` or `e`:

```js
// L97
} else if (!/[.e]/iu.test(value)) {
	// Intended to be an integer
```

For a hexadecimal literal, `e` and `E` are ordinary hex digits, not an exponent marker. So `0xe0000000000000` matches this test, skips the integer branch that compares against `Number.MAX_SAFE_INTEGER` (L98-L108), and falls through to the floating point branch (L109-L126) which only checks for subnormals. Nothing is reported.

This affects any hexadecimal literal containing `e` or `E` anywhere in its digits, not just ones that start with `0xe`. `0xdeadbeefdeadbeef` is an example where the value genuinely changes when read as a JavaScript number, and it is not reported either.

JSON5 defers its number grammar to ECMAScript 5.1, where a hexadecimal literal is a separate production from a decimal literal and cannot carry a fraction or an exponent part. So treating a hexadecimal literal as a floating point value is not correct in the first place.

**Relation to #194**

This is not the same as #194 (`fix: program crashes in no-unsafe-values`), even though that one also involved hexadecimal input. #194 fixed a crash in the `node.value === 0` branch, and added valid-case tests for zero-valued hex such as `0x0`, `0X0` and `+0x0`. The value here is not zero, so it never reaches that branch. It reaches the check at L97, which #194 did not change.

I could not find an existing issue or pull request covering this, so I am reporting it.

If this is accepted, I'd like to take it on.

**Comment by DMartens** [comment]
This seems to be an oversight that JSON5 allows hexadecimal numbers.
A PR to fix this is welcome.
